app/src/main/java/com/beemdevelopment/aegis/vault/VaultGroup.java

```diff
@@ -0,0 +1,91 @@
+package com.beemdevelopment.aegis.vault;
+
+import android.content.Context;
+
+import com.beemdevelopment.aegis.R;
+import com.beemdevelopment.aegis.util.UUIDMap;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.UUID;
+
+public class VaultGroup extends UUIDMap.Value {
+    private static final UUID NO_GROUP = UUID.fromString("00000000-0000-0000-0000-000000000000");
+
+    private String _groupName;
+
+    private VaultGroup(UUID uuid, String name) {
+        super(uuid);
+        _groupName = name;
+    }
+
+    public VaultGroup(String name) {
+        super();
+        _groupName = name;
+    }
+
+    public JSONObject toJson() {
+        JSONObject obj = new JSONObject();
+
+        try {
+            obj.put("uuid", getUUID().toString());
+            obj.put("name", _groupName);
+        } catch (JSONException e) {
+            throw new RuntimeException(e);
+        }
+
+        return obj;
+    }
+
+    public static VaultGroup fromJson(JSONObject obj) throws VaultEntryException {
+        try {
+            UUID uuid = UUID.fromString(obj.getString("uuid"));
+            String groupName = obj.getString("name");
+
+            return new VaultGroup(uuid, groupName);
+        } catch (JSONException e) {
+            throw new VaultEntryException(e);
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof VaultGroup)) {
+            return false;
+        }
+
+        VaultGroup entry = (VaultGroup) o;
+        return super.equals(entry) && equivalates(entry);
+    }
+
+    /**
+     * Reports whether this entry is equivalent to the given entry. The UUIDs of these
+     * entries are ignored during the comparison, so they are not necessarily the same
+     * instance.
+     */
+    public boolean equivalates(VaultGroup entry) {
+        return getGroupName().equals(entry.getGroupName());
+    }
+
+    public String getGroupName() {
+        return _groupName;
+    }
+
+    public void setGroupName(String groupName) {
+        _groupName = groupName;
+    }
+
+    @Override
+    public String toString() {
+        return _groupName;
+    }
+
+    public static VaultGroup getNoGroupPlaceholder(Context context) {
+        return new VaultGroup(NO_GROUP, context.getString(R.string.no_group));
+    }
+
+    public static UUID getNoGroupUuid() {
```

Comment:
The UI placeholder for 'no group' can be removed entirely once multiple groups support is added since it will be a dropdown of checkboxes and checking 0 would be equivalent. This was the plan but I felt doing it all in one PR would be too big of a change.

I've seen this approach in other codebases to put placeholder items into ViewHolders easily and having a VaultGroup object for 'no group' makes using your dropdown helpers easy. Its up to you whether you think its is too hacky to use for now. It shouldn't be a problem to just use null for filtering (which I agree seems more logical).

I had considered that a magic UUID like this could be used for things like a recycle bin or archive feature in future.